usr/sbin/registercloudguest

```diff
@@ -144,7 +144,43 @@ def setup_registry(registration_target, clean='registry'):
         utils.get_credentials_file(registration_target)
     )
     registry_fqdn = registration_target.get_registry_FQDN()
+
+    with_docker = True
+    if not utils.is_docker_present():
+        with_docker = False
+        utils.clean_registries_conf_docker(registry_fqdn)
+        logging.info(
+            'docker config file {} not found. '
+            'Registration for docker skipped/cleared. '
+            'Install docker to add docker support.'.format(
+                utils.DOCKER_CONFIG_PATH
+            )
+        )
+
     if utils.is_registry_registered(registry_fqdn):
+        if with_docker:
+            # Special handling for the docker case:
+            #
+            # When docker is present we perform the registry configuration
+            # for the docker container engine. This may cause calling the
+            # setup routine even though docker was setup before. However,
+            # the code will only effectively write a new registry setup
+            # if the required modifications are not included.
+            logging.info('Adding docker support')
+            docker_setup_ok = utils.set_registries_conf_docker(registry_fqdn)
+
+            # docker_setup_ok will take a True/False value on a real
```

Comment:
The comment is confusing. Unless I missed something it appears that `set_registries_conf_docker` does not return `True` when the config was setup successfully, it returns `None`

Reply:
set_registries_conf_docker has the following states

1. no docker config present: `None`
2. docker config present but failed to read: `False`
3. docker config present, read in and modified_by_us stays false: `None`
4. docker config present, read in and modified_by_us becomes true. This causes the call of `return write_registries_conf`. This method logs "File %s updated" and returns True: `True`
5. docker config present, read in and modified_by_us becomes true and  write_registries_conf raises: `False`

That's how I read the code

Reply:
I missed the `return` statement in front of the call to `write_registries_conf()`